_unittest/test_26_emit.py

```diff
@@ -898,6 +898,143 @@ def test_availability_1_to_1(self):
+
+        for i in range(len(radiosTX)):
+            assert tx_bands[i] == rev.get_band_names(radiosTX[i], TxRxMode.RX)[0]
+
+        # Test with no filtering
+        expected_interference_colors = [["white", "green", "yellow"], ["red", "green", "white"]]
+        expected_interference_power = [["N/A", -20.0, -20.0], [-20.0, -20.0, "N/A"]]
+        expected_protection_colors = [["white", "yellow", "yellow"], ["yellow", "yellow", "white"]]
+        expected_protection_power = [["N/A", -20.0, -20.0], [-20.0, -20.0, "N/A"]]
+
+        interference_colors = []
+        interference_power_matrix = []
+        protection_colors = []
+        protection_power_matrix = []
+        interference_colors, interference_power_matrix = interference_type_classification(self.aedtapp)
+        protection_colors, protection_power_matrix = protection_level_classification(
+            self.aedtapp, global_protection_level=True, global_levels=[30, -4, -30, -104]
+        )
+
+        assert interference_colors == expected_interference_colors
+        assert interference_power_matrix == expected_interference_power
+        assert protection_colors == expected_protection_colors
+        assert protection_power_matrix == expected_protection_power
+
+        # Test protection level with radio-specific protection levels
+        expected_protection_colors = [["white", "orange", "red"], ["yellow", "orange", "white"]]
+        expected_protection_power = [["N/A", -20.0, -20.0], [-20.0, -20.0, "N/A"]]
+        protection_levels = {
+            "Global": [30.0, -4.0, -30.0, -104.0],
+            "Bluetooth Low Energy (LE)": [30.0, -4.0, -22.0, -104.0],
+            "GPS Receiver": [30.0, -22.0, -30.0, -104.0],
+            "WiFi - 802.11-2012": [-22.0, -25.0, -30.0, -104.0],
+        }
+
+        protection_colors = []
+        protection_power_matrix = []
+        protection_colors, protection_power_matrix = protection_level_classification(
+            self.aedtapp, global_protection_level=False, protection_levels=protection_levels
+        )
+
+        assert protection_colors == expected_protection_colors
+        assert protection_power_matrix == expected_protection_power
+
+        # Test with active filtering
+        expected_interference_colors = [["white", "green", "yellow"], ["white", "green", "white"]]
+        expected_interference_power = [["N/A", -20.0, -20.0], ["<= -200", -20.0, "N/A"]]
+        expected_protection_colors = [["white", "white", "white"], ["white", "white", "white"]]
+        expected_protection_power = [["N/A", "< -200", "< -200"], ["< -200", "< -200", "N/A"]]
+
+        interference_colors = []
+        interference_power_matrix = []
+        protection_colors = []
+        protection_power_matrix = []
+        interference_filter = [
+            "TxFundamental:Out-of-band",
+            ["TxHarmonic/Spurious:Out-of-band", "Intermod:Out-of-band", "Broadband:Out-of-band"],
+        ]
```

Comment:
Can you explain this? 
![image](https://github.com/ansys/pyaedt/assets/101877396/4b1075bb-e1b5-4fc0-b174-b80fc3744c3a)

From this picture, you want to do only for TxFundamental:Out-of-band?

Reply:
The gui/script classifies interference slightly different that By Problem Type. For the source side, it checks whether the signal is fundamental or not. On the victim side, it checks if the source signal falls within the band. For each case of interference, a string with the format of those in "interference_filter" is created and checked against the list of filter values to check if the result is within the scope of the filters.